packages/netlify-cms-backend-github/src/API.js

```diff
@@ -94,80 +95,50 @@ export default class API {
-    return this.request(`${this.repoURL}/git/refs/meta/_netlify_cms?${Date.now()}`, {
+  generateContentKey(collection, slug) {
+    return `${collection}-${slug}`;
+  }
+
+  checkStatusLabel(status) {
+    const labelName = this.statusLabels.getIn([status, 'name']);
+
+    return this.request(`${this.repoURL}/labels/${labelName}?${Date.now()}`, {
       cache: 'no-store',
     })
-      .then(response => response.object)
+      .then(response => response.name)
       .catch(() => {
-        // Meta ref doesn't exist
-        const readme = {
-          raw:
-            '# Netlify CMS\n\nThis tree is used by the Netlify CMS to store metadata information for specific files and branches.',
-        };
-
-        return this.uploadBlob(readme)
-          .then(item =>
-            this.request(`${this.repoURL}/git/trees`, {
-              method: 'POST',
-              body: JSON.stringify({
-                tree: [{ path: 'README.md', mode: '100644', type: 'blob', sha: item.sha }],
-              }),
-            }),
-          )
-          .then(tree => this.commit('First Commit', tree))
-          .then(response => this.createRef('meta', '_netlify_cms', response.sha))
-          .then(response => response.object);
+        // Label doesn't exist, create label
+        return this.request(`${this.repoURL}/labels`, {
+          method: 'POST',
+          body: JSON.stringify({
+            name: labelName,
+            color: this.statusLabels.getIn([status, 'color']),
+          }),
+        }).then(response => response.name);
       });
   }
 
-  storeMetadata(key, data) {
-    return this.checkMetadataRef().then(branchData => {
-      const fileTree = {
-        [`${key}.json`]: {
-          path: `${key}.json`,
-          raw: JSON.stringify(data),
-          file: true,
-        },
-      };
-
-      return this.uploadBlob(fileTree[`${key}.json`])
-        .then(() => this.updateTree(branchData.sha, '/', fileTree))
-        .then(changeTree => this.commit(`Updating “${key}” metadata`, changeTree))
-        .then(response => this.patchRef('meta', '_netlify_cms', response.sha))
-        .then(() => {
-          localForage.setItem(`gh.meta.${key}`, {
-            expires: Date.now() + 300000, // In 5 minutes
-            data,
-          });
-        });
-    });
-  }
-
-  retrieveMetadata(key) {
-    const cache = localForage.getItem(`gh.meta.${key}`);
-    return cache.then(cached => {
-      if (cached && cached.expires > Date.now()) {
-        return cached.data;
-      }
-      console.log(
-        '%c Checking for MetaData files',
-        'line-height: 30px;text-align: center;font-weight: bold',
-      );
-      return this.request(`${this.repoURL}/contents/${key}.json`, {
-        params: { ref: 'refs/meta/_netlify_cms' },
-        headers: { Accept: 'application/vnd.github.VERSION.raw' },
-        cache: 'no-store',
+  retrievePrData(branchName) {
+    let pr;
+    return this.request(`${this.repoURL}/pulls`, {
```

Comment:
`pulls` is a paginated endpoint: https://developer.github.com/v3/pulls/#response